src/v/kafka/server/handlers/topics/types.cc

```diff
@@ -31,7 +31,12 @@
 
 namespace kafka {
 
-config_map_t config_map(const std::vector<createable_topic_config>& config) {
+template<typename T>
+concept CreatableTopicCfg = std::is_same_v<T, creatable_topic_configs> || std::
+  is_same_v<T, createable_topic_config>;
```

Comment:
Ah I think that was done on purpose. If you look here at upstream kafka ( [request](https://github.com/apache/kafka/blob/trunk/clients/src/main/resources/common/message/CreateTopicsRequest.json#L50), [response](https://github.com/apache/kafka/blob/trunk/clients/src/main/resources/common/message/CreateTopicsResponse.json#L55) ) you'll see it originated from there.

I think it was done this way on purpose because when the generator runs through these source files it will create a uniquely named struct for each field. A name collision would occur if these names were not unique.

Reply:
🙈